src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -392,7 +392,8 @@ protected function initializeContainer(): void
     {
         $class = $this->getContainerClass();
         $buildDir = $this->warmupDir ?: $this->getBuildDir();
-        $cache = new ConfigCache($buildDir.'/'.$class.'.php', $this->debug);
+        $cache = new ConfigCache($buildDir.'/'.$class.'.php', $this->debug && !($_SERVER['SYMFONY_DISABLE_RESOURCE_TRACKING'] ?? false));
```

Comment:
This env var might not be enough: an external process won't be able to replace all self-checking resources. It could only replace those it knows about, and since projects can create their own resource types, we'll have to evaluate them in PHP.
Instead of a boolean, this env var could be a list of strings: the class names the external process knows about. The PHP process would then skip the listed ones.

Reply:
PR updated with this suggestion :)